bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/servlets/MultiFieldDropTargetPostProcessor.java

```diff
@@ -0,0 +1,155 @@
+ * 
+ *    Copyright (c) 2018 iDA MediaFoundry
+ * 
+ *    Permission is hereby granted, free of charge, to any person obtaining a
+ *    copy of this software and associated documentation files (the "Software"),
+ *    to deal in the Software without restriction, including without limitation
+ *    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ *    and/or sell copies of the Software, and to permit persons to whom the Software
+ *    is furnished to do so, subject to the following conditions:
+ * 
+ *    The above copyright notice and this permission notice shall be included in all
+ *    copies or substantial portions of the Software.
+ * 
+ *    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *    WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *    CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ *******************************************************************************/
+
+package com.adobe.cq.commerce.core.components.internal.servlets;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.request.RequestParameter;
+import org.apache.sling.api.request.RequestParameterMap;
+import org.apache.sling.api.resource.ModifiableValueMap;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.servlets.post.Modification;
+import org.apache.sling.servlets.post.SlingPostProcessor;
+import org.osgi.service.component.annotations.Component;
+
+import com.google.common.collect.Iterators;
+
+/**
+ * This class is originally available at <code>https://github.com/ida-mediafoundry/jetpack-advanced-drop-targets</code><br>
+ * <br>
+ * This PostProcessor listens to all post servlet actions that contain a property that starts with <code>"./dropTarget->"</code>.
+ * It will take the value from the property and add it to the property that follows the drop target prefix.<br>
+ * <br>
+ * In the following example, it will take the value from "./dropTarget->@books" and add it to the property "books"
+ * 
+ * <pre>
+ * {@literal
+ * <cq:dropTargets jcr:primaryType="nt:unstructured">
+ *   <books
+ *      jcr:primaryType="cq:DropTargetConfig"
+ *      ...
+ *      propertyName="./dropTarget->@books">
+ *   </books>
+ * </cq:dropTargets>
+ * }
+ * </pre>
+ * 
+ */
+@Component(service = { SlingPostProcessor.class })
+public class MultiFieldDropTargetPostProcessor implements SlingPostProcessor {
```

Comment:
Would it possible to scope this servlet somehow to our CIF components? I'm wondering what would happen if a customer already has a custom drop target servlet like this for other components?

At least we should mention this in the documentation or component readme.

Reply:
It's kind of scoped because it only handles property keys starting with `./dropTarget->` so it could only "collide with itself" if a customer already uses the same post processor. I will extend the documentation or component readme.